infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -339,11 +339,26 @@ private synchronized void runDetectionTasks() {
             // Corrects out of phase epoch issues if present in the report. This method
             // performs re-sealing of all nodes if required and catchup of a layout server to
             // the current state.
-            Layout updatedLayout = correctWrongEpochs(pollReport, layout);
+            final Layout updatedLayout = correctWrongEpochs(pollReport, layout);
 
             Result<DetectorTask, RuntimeException> failure = Result.of(() -> {
-                if (pollReport.isCurrentLayoutSlotUnFilled()) {
-                    detectFailure(updatedLayout, Collections.emptySet(), pollReport).join();
+
+                // This is just an optimization in case we receive a WrongClusterException
+                // while one of the other management clients is trying to move to a new layout.
+                // This check is merely trying to minimize the the scenario in which we end ip
+                // filling the slot with an outdated layout.
+                if (!pollReport.areAllResponsiveServersSealed()) {
```

Comment:
The check and the log are counter-intuitive. We enter the if statement if all responsive servers are not sealed. However, we log that all servers have been sealed.
Is this logic correct?

Reply:
You should take a look at the latest patch and see if it makes more sense.

Reply:
I see now. The message is wrong. Fixed.